src/coreclr/nativeaot/Runtime/amd64/MiscStubs.asm

```diff
@@ -41,37 +39,18 @@ LEAF_END RhpStackProbe, _TEXT
 
 LEAF_ENTRY RhpGetThreadStaticBaseForType, _TEXT
```

Comment:
The "fast" access got a lot simpler here. `RhpGetThreadStaticBaseForType` is what we want eventually not called, but directly inlined by the JIT into callers.

We could yet make this a bit simpler by removing an indirection into the array of storage instances. It may be somewhat challenging though.

Reply:
Is there any other technical reason why we don't have such assembly helper for Arm64 apart that it is not implemented?

Reply:
The helper for Arm64 is now implemented